packages/cozy-harvest-lib/src/models/ConnectionFlow.js

```diff
@@ -617,20 +655,45 @@ export class ConnectionFlow {
    * If the trigger we display is already running, subscribe to the associated job
    */
   watchCurrentJobIfTriggerIsAlreadyRunning() {
-    if (get(this, 'trigger.current_state.status') === 'running') {
+    if (this.trigger?.current_state?.status === 'running') {
       this.job = {
-        _id: get(this, 'trigger.current_state.last_executed_job_id')
+        _id: this.trigger?.current_state?.last_executed_job_id
       }
       this.watchJob()
     }
   }
 
+  /**
+   * Watch all jobs related to the current trigger
+   */
+  watchTriggerJobs() {
+    if (this.trigger) {
+      // When the trigger comes from realtime, cozy-stack does not add the current_state to the object. So we need to request the stack to get it
+      if (!this.trigger?.current_state) {
+        this.refetchTrigger()
+      }
```

Comment:
It seems that `watchTriggerJobs` is only called from the constructor: https://github.com/cozy/cozy-libs/blob/5e25cd6da63b93558bf3609d4230153097476ca8/packages/cozy-harvest-lib/src/models/ConnectionFlow.js#LL185C18-L185C18 

How can we have a trigger without `current_state` at this point?  Can you explain how it can happen?

Reply:
Is it because when it's the first run of the trigger, then we don't have yet a `current_state`?

Reply:
This trigger instance comes from the host application of harvest (home for example) which can get it from a realtime event, which does not harvest current_state